packages/react-components/src/components/knowledge-graph/KnowledgeGraphPanel.tsx

```diff
@@ -72,20 +105,26 @@ export const KnowledgeGraphContainer: React.FC<KnowledgeGraphInterface> = ({
     });
   }, []);
 
+  // using async await here is a temprory Hack for the case when you select a node and then click another node,
+  // in such a case Cytoscape fires the select event first before the unselect event of the first node.
+  // We are also considering another solution mentioned here
+  // https://stackoverflow.com/questions/61898195/find-selected-nodes-edges-in-cytoscape-js-onunselect-event
+
   const clickEntityHandler = useCallback(
-    ({ target }: EventObjectNode) => {
-      const data = target.data() as NodeData;
+    async ({ target }: EventObjectNode) => {
+      const data = (await target.data()) as NodeData;
       if (onEntitySelected) {
         onEntitySelected(data);
       }
       setSelectedGraphNodeEntityId(data.id);
+      console.log('selectedGraphNodeEntityId Inside = ', selectedGraphNodeEntityId);
     },
     [onEntitySelected, setSelectedGraphNodeEntityId]
   );
 
   const clickRelationshipHandler = useCallback(
-    ({ target }: EventObjectEdge) => {
-      const data = target.data() as EdgeData;
+    async ({ target }: EventObjectEdge) => {
+      const data = (await target.data()) as EdgeData;
```

Comment:
Using async await here is a temporary hack to address a corner case, see the comment and the stackoverflow link in the code for more explanation.